From da61dedfa523f3e6761ed0da91d8d95c6d4caebd Mon Sep 17 00:00:00 2001
From: Colin Walters <walters@verbum.org>
Date: Sat, 14 Jan 2012 16:33:56 -0500
Subject: [PATCH] configure: Support building without documentation

Docbook is the source of many dependency cycles; allow building
without it for bootstrapping.
---
 configure.ac        |   12 +++++++++---
 doc/man/Makefile.am |    6 +++++-
 2 files changed, 14 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index a794bfc..d906d05 100644
--- a/configure.ac
+++ b/configure.ac
@@ -24,9 +24,15 @@ GNOME_DEBUG_CHECK
 GNOME_COMPILE_WARNINGS([maximum])
 GNOME_MAINTAINER_MODE_DEFINES
 
-AC_PATH_PROG([XSLTPROC], [xsltproc])
-if test -z "$XSLTPROC"; then
-  AC_MSG_ERROR([xsltproc is needed])
+AC_ARG_ENABLE([documentation],
+	AS_HELP_STRING([--enable-documentation], [enable man pages and HTML]),
+	[], [enable_documentation=yes])
+AM_CONDITIONAL(ENABLE_DOCUMENTATION, test x$enable_documentation = xyes)
+if test x$enable_documentation = xyes; then
+  AC_PATH_PROG([XSLTPROC], [xsltproc])
+  if test -z "$XSLTPROC"; then
+    AC_MSG_ERROR([xsltproc is needed for documentation])
+  fi
 fi
 
 GTK_DOC_CHECK([1.3])
diff --git a/doc/man/Makefile.am b/doc/man/Makefile.am
index 5e4946d..a03418f 100644
--- a/doc/man/Makefile.am
+++ b/doc/man/Makefile.am
@@ -1,11 +1,15 @@
 
 NULL =
 
-man_MANS = 				\
+man_MANS =
+
+if ENABLE_DOCUMENTATION
+man_MANS +=				\
 	udisksctl.1			\
 	udisksd.8			\
 	udisks.8			\
 	$(NULL)
+endif
 
 udisksctl.1 : udisksctl.xml
 	$(XSLTPROC) -nonet http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl $<
-- 
1.7.6.4

